posthog/models/cohort.py

```diff
@@ -146,15 +146,14 @@ def calculate_people_ch(self):
 
             try:
                 recalculate_cohortpeople(self)
-                self.is_calculating = False
                 self.last_calculation = timezone.now()
                 self.errors_calculating = 0
-                self.save()
-            except Exception as err:
-                self.is_calculating = False
+            except Exception as e:
                 self.errors_calculating = F("errors_calculating") + 1
+                raise e
+            finally:
+                self.is_calculating = False
```

Comment:
unrelated bufgix?

Do you know why we previously swallowed the error?

Reply:
It's related in that swallowing the error here made it opaque why a test case failed. And I think it should be clear if anything breaks in the future.

Reply:
`calculate_people` behaves subtly different now though - if we wanted to remove this behavior, worth removing it in both places. Though scope creep :sweat_smile:

Reply:
Good catch with `calculate_people`. As for swallowing in the task, I don't really see the point in that. Celery already doesn't crash on errors, and the error notice it leaves in stdout is possibly useful.